File: frontend/src/types/trip.ts
import {Participant} from "@/types/participant";
import {Activity} from "@/types/activity";
import {Expense} from "@/types/expense";
import {ChecklistItem} from "@/types/checklist";

export interface Trip {
    id: string
    title: string
    description: string
    startDate: string // ISO date string (YYYY-MM-DD)
    endDate: string // ISO date string (YYYY-MM-DD)
    participantCount: number
}

export interface TripDetail extends Trip {
    participants: Participant[]
    activities: Activity[]
    expenses: Expense[]
    checklist: ChecklistItem[]
}


export interface CreateTripRequest {

    title: string;

    description: string;

    startDate: string;

    endDate: string;

}

File: Backend/src/main/java/JourneySync/demo/entity/Trip.java
package JourneySync.demo.entity;

import jakarta.persistence.*;
import lombok.*;


import java.time.LocalDate;
import java.util.List;
import java.util.UUID;
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder

@Entity
@Table(name = "trips")
public class Trip {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    private String title;
    private String description;

    private LocalDate startDate;
    private LocalDate endDate;

    @OneToMany(mappedBy = "trip", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Participant> participants;

    @OneToMany(mappedBy = "trip", cascade = CascadeType.ALL, orphanRemoval = true)
    private List<Activity> activities;
}
